packages/whenable/package.json

```diff
@@ -0,0 +1,50 @@
+{
+  "name": "@agoric/whenable",
+  "version": "0.1.0",
+  "description": "Remote (shortening and disconnection-tolerant) Promise-likes",
+  "type": "module",
+  "main": "src/index.js",
+  "engines": {
+    "node": ">=14.15.0"
+  },
+  "scripts": {
+    "build": "exit 0",
+    "prepack": "tsc --build tsconfig.build.json",
+    "postpack": "git clean -f '*.d.ts*'",
+    "test": "ava",
+    "test:nyc": "exit 0",
+    "test:xs": "exit 0",
+    "lint-fix": "yarn lint:eslint --fix",
+    "lint": "run-s --continue-on-error lint:*",
+    "lint:eslint": "eslint .",
+    "lint:types": "tsc"
+  },
+  "dependencies": {
+    "@agoric/base-zone": "^0.1.0",
+    "@agoric/internal": "^0.3.2",
+    "@endo/far": "^1.0.1",
+    "@endo/pass-style": "^1.0.1",
+    "@endo/patterns": "^1.0.1",
+    "@endo/promise-kit": "^1.0.1"
+  },
+  "devDependencies": {
+    "@endo/init": "^1.0.1",
+    "ava": "^5.3.0"
+  },
```

Comment:
> knowing what a `DisconnectionObject` is

To preserve the correct layering, I'd put a `prepareWhenableModule` wrapper in `@agoric/internal`, have it close over the `DisconnectionObject`.

>  Could we make `E` work with whenable

I'd want to have the base `prepareWhenableModule` also return an `E` that was constructed via `makeE` from `@endo/eventual-send/src/E.js`.  There would be a little bit of work to generalize `makeE` to accept the necessary powers (similar to the ones in my eventual-send revamp slides).

Reply:
As for `E.when(specimen, ..., ...)`, it has exactly the interface (by design) as the `when` operator returned by `prepareWhenableModule`.

I'd also introduce `E.watch(specimen, watcher): Whenable` to use the `watch` operator, which (unlike `E.when`) always returns a Whenable.  It also works across upgrade incarnations, presuming that if the initial `specimen` is a Promise, it settles promptly (and its fulfilment can be a Whenable).

Reply:
I like moving `watch` into `E.watch`. I also like `watch` aligning with `observe` as in the `observeFoo(foo, fooObserver)` used by @agoric/notifier. In retrospect I wish we had gone with `watchFoo(foo, fooWatcher)`, as it is shorter and no less clear. Nevertheless, since we didn't, having the unqualified `watch` and `Watcher` be about promise-likes seems a good choice, considering how much more pervasive promises are than any other observable.

Reply:
Resolving